### DISCUSSION TITLE
Design improvement discussion - Chat

### DISCUSSION THREAD

**Discussion by dzuk-mutant** [datafruits/datafruits#1350] [Ideas]
Starting with thoughts about the current user experience as usual! I have a few sketches for an updated design for this, but I don't have the time to flesh them out right now!

### Music player
I think it could be a more integrated part of the experience. This overlaps with the nav discussion.
- It would be helpful if there's a way to know what show is up next and how long until it starts.
- Hardcoded dates on live shows creates conflicts of information with different time zones. For instance, I was confused about a live show that said 12th July when it was 13th of July in the UK. Generally I think the music player metadata should follow from the timetable information and archive metadata, not track metadata.

I think my idea of breaking down the main navigation areas into widgets is advantageous here, as it's possible to move the music player to a more relevant part of the interface when a show is going on.

### User area
- It would be cool if the user list could explicitly point out who the DJs and VJs are.
- I think long usernames could be truncated or overflowed onto a new line.

### Chat
- I think that the chat text colour and style could be calibrated to work in a more universal way. Giving a colour picker is not the best user experience, and there is not one setting that provides a reasonable amount of contrast in all situations, so the colour has to be re-picked as a show goes on, or when a different show comes up, or when viz stops.
- I think the images/GIFs could be smaller and be plenty visible while taking less attention away from viz. Tall GIFs especially really dominate the view.
- I also think text areas should be made narrower. Wide text areas are harder to read and overlap viz more than they need to.
- Also also, I think the message bar could easily be a lot narrower. It takes up more space than necessary on desktop.
- IMO, seconds are unnecessary in the timestamp. The user can already infer that one message is earlier than another if they're within the same minute of each other.
- Text styling and layout of each component (timestamp, avatar, username, message) could be tweaked to provide more information hierarchy and flow.

![image](https://github.com/user-attachments/assets/cd927093-e3e9-4e21-b480-14cf8d26d2ab)

Generally speaking, I think there's a lack of compositional balance going on in this area, and I think the parts could be tuned to be more harmonious with viz and be easier to read with little to no effort on the user's part.

### Things I feel less strongly about but could be a cool idea:
- Perhaps when viz is on, the chat colour should be more colour neutral to get out of the way of the video more.
- A GIF/fruits/emoji picker that doesn't rely on a floating window might be cool because there's never a hugely convenient place to just have it on the screen, it's usually overlapping something.

**Comment by dudethatbe** [comment]
I appreciate the detailed notes as always @dzuk-mutant! Wish I could +1 each point, so just assume those little emojis are embedded in there. Here's some context on a few points:

## Showing what's live on music player
Regardless of what improvements are made on the frontend, there'll have to be some bug fixes in the backend to really fix this. For instance, the text that display "LIVE -- oven" is sometimes wrong and shows the name of the _last_ DJ that played. Or sometimes the DJ is disconnected, but the text still says LIVE (while showing the sparkling animation)... 

That's all to say, yes we should have an indicator that says "This is a live show". I wonder how we should indicate that it's a pre-recorded show though. Is it necessary to say this isn't strictly live? It would be interesting to see something like "[LIVE] [PRE_RECORD] Good Morning Datafruits - ovenrake", but something shorter that PRE_RECORD 🙃 

## User area labels
I think the last time we had a thought about this was when we created this ticket with no explanation https://github.com/datafruits/datafruits/issues/1279
We technically have the badges that say DJ/VJ but nothing to say who are the _active_ artists at that moment. Which would be really cool to show so I really like this idea. I think once we finally have user-auth for viz, then we can track who the dj/vjs are and make that available as an API endpoint (or websocket event).

## Chat text styles
This always used to be one color and during a weekend marathon of DJs and VJs, someone suggested that clicking on the text should change the color (no additional comments lol). Since it's hard to anticipate what the viz is displaying and what the active theme is, it doesn't seem realistic to achieve optimal contrast. We sought out to improve it and here's where we started (https://github.com/datafruits/datafruits/issues/449) note the unchecked **Contrast** section.
Since then I got into learning about what are acceptable levels of contrast and ending up adjusting the 4 main colors of the site https://github.com/datafruits/datafruits/pull/1041/files. The text contrast problem remained so I finally just added a color-picker, but maybe I'm the only one that actually likes the feature 😅

The controversial thing to suggest here is to not overlay the chat over viz. Personally I like it despite the poor contrast, but I support the choice to have it in a dedicated area for optimal contrast. Maybe we can default it to a normal chat component, and have a docking option that sets it to overlay the viz?
